sql/src/main/java/io/crate/operation/collect/EngineSearcher.java

```diff
@@ -39,9 +43,12 @@
             try {
                 engineSearcher = indexShard.acquireSearcher(searcherName);
             } catch (IllegalIndexShardStateException e) {
-                if (e.currentState() == IndexShardState.POST_RECOVERY && retry < 10) {
+                if (e.currentState() == IndexShardState.POST_RECOVERY && retry < 100) {
                     try {
-                        Thread.sleep(retry);
+                        if (LOGGER.isWarnEnabled() && retry > 10) {
+                            LOGGER.warn("shard in POST_RECOVERY - retry: " + retry);
```

Comment:
yep, if the retry count is above 10 that means the cluster is already really busy. In local tests with 100% cpu and lots of partitions being created the avg was at ~4